From bcbdd2b58b534b77be2f35bb8d38bbe5cc05e85f Mon Sep 17 00:00:00 2001
From: Heinrich Toews <heinrich.toews@wago.com>
Date: Wed, 26 Jul 2017 17:20:52 +0000
Subject: [PATCH] cpsw: increase rx_packet_max by 4 bytes

Due to a special dsa marvell tagging it is mandatory to
increase the maximum rx packet size by 4 bytes.
Otherwise the CPSW switching engine is dropping the packets.

As it is only a MV related issue may be it would be better to
use the commandline paramter 'rx_packet_max=<size>' to adjust
the max size.
---
 drivers/net/ethernet/ti/cpsw.c | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/drivers/net/ethernet/ti/cpsw.c b/drivers/net/ethernet/ti/cpsw.c
index d7cb205f..8c790ff 100644
--- a/drivers/net/ethernet/ti/cpsw.c
+++ b/drivers/net/ethernet/ti/cpsw.c
@@ -120,7 +120,8 @@ do {								\
 
 #define CPSW_POLL_WEIGHT	64
 #define CPSW_MIN_PACKET_SIZE	60
-#define CPSW_MAX_PACKET_SIZE	(1500 + 14 + 4 + 4)
+#define CPSW_MARVELL_TAG_SIZE   4
+#define CPSW_MAX_PACKET_SIZE	(1500 + 14 + 4 + 4 + CPSW_MARVELL_TAG_SIZE)
 
 #define RX_PRIORITY_MAPPING	0x76543210
 #define TX_PRIORITY_MAPPING	0x33221100
@@ -1186,6 +1187,8 @@ static void cpsw_slave_open(struct cpsw_slave *slave, struct cpsw_priv *priv)
 		break;
 	}
 
+        pr_info("CPSW: Setting rx_packet_max to %d\n", cpsw->rx_packet_max);
+
 	/* setup max packet size, and mac address */
 	__raw_writel(cpsw->rx_packet_max, &slave->sliver->rx_maxlen);
 	cpsw_set_slave_mac(slave, priv);
-- 
2.7.4

